app/Keycloak/TokenStrategy/ClientCredentials.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Keycloak\TokenStrategy;
+
+use App\Json;
+use App\Keycloak\Dto\Config;
+use App\Keycloak\Dto\Realm;
+use App\Keycloak\Exception\KeyCloakApiFailed;
+use GuzzleHttp\ClientInterface;
+use GuzzleHttp\Exception\GuzzleException;
+
+/*
+ * LIMITATION: This class currently does not refresh the token automatically.
+ * For normal usage this should be ok, but if we ever implement long-running CLI processes this will need to be improved.
+ * */
+final class ClientCredentials implements TokenStrategy
+{
+    private ?string $accessToken = null;
+
+    public function __construct(
+        private readonly ClientInterface $client,
+        private readonly Config $config
+    ) {
+    }
+
+    public function fetchToken(Realm $realm): string
+    {
+        if ($this->accessToken === null) { //lazy loading
```

Comment:
this can become an early return instead

Reply:
I'm also not sure if this approach is good. What if you change the realm? Then you still get the token of the previous realm.